### ISSUE TITLE
travis-ci build failures

### ISSUE THREAD

**Issue by chipitsine** [haproxy/haproxy#1591]
### Tool Name and Version

gcc

### Code Report

```plain
s390x:

https://app.travis-ci.com/github/haproxy/haproxy/jobs/561325455


  CC      src/server.o

src/stream.c: In function ‘cli_io_handler_dump_sess’:

src/stream.c:3643:24: error: null pointer dereference [-Werror=null-dereference]

 3643 |          curr_strm->csf->si->state,

      |          ~~~~~~~~~~~~~~^~~~

src/stream.c:3653:24: error: null pointer dereference [-Werror=null-dereference]

 3653 |          curr_strm->csb->si->state,

      |          ~~~~~~~~~~~~~~^~~~

  CC      src/flt_spoe.o

  CC      src/stick_table.o

src/stats.c: In function ‘stats_dump_stat_to_buffer’:

src/stats.c:3444:34: error: potential null pointer dereference [-Werror=null-dereference]

 3444 |                (appctx->ctx.stats.flags & STAT_HIDEVER) ? "" : (stats_version_string),

      |                 ~~~~~~~~~~~~~~~~~^~~~~~

src/stats.c:3444:34: error: potential null pointer dereference [-Werror=null-dereference]

src/stream.c: At top level:

cc1: error: unrecognized command line option ‘-Wno-atomic-alignment’ [-Werror]

cc1: error: unrecognized command line option ‘-Wno-string-plus-int’ [-Werror]

cc1: all warnings being treated as errors

make: *** [Makefile:984: src/stream.o] Error 1

make: *** Waiting for unfinished jobs....

src/stats.c: At top level:

cc1: error: unrecognized command line option ‘-Wno-atomic-alignment’ [-Werror]

cc1: error: unrecognized command line option ‘-Wno-string-plus-int’ [-Werror]

cc1: all warnings being treated as errors

make: *** [Makefile:984: src/stats.o] Error 1
```

arm64-graviton2
https://app.travis-ci.com/github/haproxy/haproxy/jobs/561325454

```
  CC      src/server.o

src/stream.c: In function ‘cli_io_handler_dump_sess’:

src/stream.c:3643:24: error: null pointer dereference [-Werror=null-dereference]

 3643 |          curr_strm->csf->si->state,

      |          ~~~~~~~~~~~~~~^~~~

src/stream.c:3653:24: error: null pointer dereference [-Werror=null-dereference]

 3653 |          curr_strm->csb->si->state,

      |          ~~~~~~~~~~~~~~^~~~

src/stats.c: In function ‘stats_dump_stat_to_buffer’:

src/stats.c:3444:34: error: potential null pointer dereference [-Werror=null-dereference]

 3444 |                (appctx->ctx.stats.flags & STAT_HIDEVER) ? "" : (stats_version_string),

      |                 ~~~~~~~~~~~~~~~~~^~~~~~

src/stats.c:3444:34: error: potential null pointer dereference [-Werror=null-dereference]

src/stream.c: At top level:

cc1: error: unrecognized command line option ‘-Wno-atomic-alignment’ [-Werror]

cc1: error: unrecognized command line option ‘-Wno-string-plus-int’ [-Werror]

cc1: all warnings being treated as errors

make: *** [Makefile:984: src/stream.o] Error 1

make: *** Waiting for unfinished jobs....

src/stats.c: At top level:

cc1: error: unrecognized command line option ‘-Wno-atomic-alignment’ [-Werror]

cc1: error: unrecognized command line option ‘-Wno-string-plus-int’ [-Werror]

cc1: all warnings being treated as errors

make: *** [Makefile:984: src/stats.o] Error 1
```

ppc64le
https://app.travis-ci.com/github/haproxy/haproxy/jobs/561325453

```
  CC      src/stats.o

src/stream.c: In function ‘cli_io_handler_dump_sess’:

src/stream.c:3643:24: error: null pointer dereference [-Werror=null-dereference]

 3643 |          curr_strm->csf->si->state,

      |          ~~~~~~~~~~~~~~^~~~

src/stream.c:3653:24: error: null pointer dereference [-Werror=null-dereference]

 3653 |          curr_strm->csb->si->state,

      |          ~~~~~~~~~~~~~~^~~~

  CC      src/server.o

  CC      src/flt_spoe.o

  CC      src/stick_table.o

  CC      src/tools.o

src/stream.c: At top level:

cc1: error: unrecognized command line option ‘-Wno-atomic-alignment’ [-Werror]

cc1: error: unrecognized command line option ‘-Wno-string-plus-int’ [-Werror]

cc1: all warnings being treated as errors

make: *** [Makefile:984: src/stream.o] Error 1

make: *** Waiting for unfinished jobs....

src/stats.c: In function ‘stats_dump_stat_to_buffer’:

src/stats.c:3444:34: error: potential null pointer dereference [-Werror=null-dereference]

 3444 |                (appctx->ctx.stats.flags & STAT_HIDEVER) ? "" : (stats_version_string),

      |                 ~~~~~~~~~~~~~~~~~^~~~~~

src/stats.c:3444:34: error: potential null pointer dereference [-Werror=null-dereference]

src/stats.c: At top level:

cc1: error: unrecognized command line option ‘-Wno-atomic-alignment’ [-Werror]

cc1: error: unrecognized command line option ‘-Wno-string-plus-int’ [-Werror]

cc1: all warnings being treated as errors

make: *** [Makefile:984: src/stats.o] Error 1
```
```


### Additional Information

_No response_

### Output of `haproxy -vv`

```plain
no
```

**Comment by capflam** [comment]
It should be fixed by 693b23bb1

**Comment by chipitsine** [comment]
let's wait for weekly build (I'll reopen if it will fail)
